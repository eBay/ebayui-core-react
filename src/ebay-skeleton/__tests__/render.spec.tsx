/// <reference types="@testing-library/jest-dom" />
import React, { ReactElement } from 'react'
import { render, screen } from '@testing-library/react'
import { composeStories } from '@storybook/react'
import * as stories from './index.stories'

const {
    Default,
    Avatar,
    Button,
    Text,
    Composite,
    Image,
    Tile,
    TextBox,
} = composeStories(stories)


describe('Render <EbaySkeleton /> component', () => {
    it.each([
        ['EbaySkeleton', Default, {}],
        ['EbaySkeleton color purple', Default, { color: 'purple' }],
        ['EbaySkeleton color green', Default, { color: 'green' }],
        ['EbaySkeleton color blue', Default, { color: 'blue' }],
        ['EbaySkeletonAvatar', Avatar, {}],
        ['EbaySkeletonAvatar with "as" prop', Avatar, { as: 'span' }],
        ['EbaySkeletonButton', Button, {}],
        ['EbaySkeletonButton with size prop large', Button, { size: 'large' }],
        ['EbaySkeletonButton with size prop small', Button, { size: 'small' }],
        ['EbaySkeletonButton with "as" prop', Button, { as: 'span' }],
        ['EbaySkeletonText', Text, {}],
        ['EbaySkeletonText with multiline prop', Text, { multiline: true }],
        ['EbaySkeletonText with size prop', Text, { size: 'large' }],
        ['EbaySkeletonText with "as" prop', Text, { as: 'span' }],
        ['EbaySkeleton composite', Composite, {}],
        ['EbaySkeletonImage', Image, {}],
        ['EbaySkeletonImage with "as" prop', Image, { as: 'span' }],
        ['EbaySkeleton tile', Tile, {}],
        ['EbaySkeletonTextBox', TextBox, {}],
        ['EbaySkeletonTextBox with "as" prop', TextBox, { as: 'span' }],
    ])(`should render the %s component`, (_, Component, props) => {
        render(<Component {...(props as stories.EbaySkeletonStoriesProps)} />)
        expect(screen.getByLabelText('Loading...')).toMatchSnapshot()
    })
})
