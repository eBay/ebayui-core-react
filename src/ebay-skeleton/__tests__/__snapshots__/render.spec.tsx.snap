// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render <EbaySkeleton /> component should render the EbaySkeleton color blue component 1`] = `
<div
  aria-label="Loading..."
  class="skeleton skeleton--blue"
  role="img"
  style="width: 220px;"
>
  <div
    class="skeleton__textbox"
  />
</div>
`;

exports[`Render <EbaySkeleton /> component should render the EbaySkeleton color green component 1`] = `
<div
  aria-label="Loading..."
  class="skeleton skeleton--green"
  role="img"
  style="width: 220px;"
>
  <div
    class="skeleton__textbox"
  />
</div>
`;

exports[`Render <EbaySkeleton /> component should render the EbaySkeleton color purple component 1`] = `
<div
  aria-label="Loading..."
  class="skeleton skeleton--purple"
  role="img"
  style="width: 220px;"
>
  <div
    class="skeleton__textbox"
  />
</div>
`;

exports[`Render <EbaySkeleton /> component should render the EbaySkeleton component 1`] = `
<div
  aria-label="Loading..."
  class="skeleton"
  role="img"
  style="width: 220px;"
>
  <div
    class="skeleton__textbox"
  />
</div>
`;

exports[`Render <EbaySkeleton /> component should render the EbaySkeleton composite component 1`] = `
<div
  aria-label="Loading..."
  class="skeleton"
  role="img"
  style="width: 300px;"
>
  <div>
    <span
      class="skeleton__avatar"
    />
    <span
      class="skeleton__text skeleton__text--multiline"
      style="width: 220px; vertical-align: top;"
    />
  </div>
  <div
    class="skeleton__button"
  />
</div>
`;

exports[`Render <EbaySkeleton /> component should render the EbaySkeleton tile component 1`] = `
<div
  aria-label="Loading..."
  class="skeleton"
  role="img"
  style="width: 220px;"
>
  <div
    class="skeleton__image"
    style="width: 220px; height: 220px;"
  />
  <div
    class="skeleton__text skeleton__text--multiline skeleton__text--large"
  />
</div>
`;

exports[`Render <EbaySkeleton /> component should render the EbaySkeletonAvatar component 1`] = `
<div
  aria-label="Loading..."
  class="skeleton"
  role="img"
>
  <div
    class="skeleton__avatar"
  />
</div>
`;

exports[`Render <EbaySkeleton /> component should render the EbaySkeletonAvatar with "as" prop component 1`] = `
<div
  aria-label="Loading..."
  class="skeleton"
  role="img"
>
  <span
    class="skeleton__avatar"
  />
</div>
`;

exports[`Render <EbaySkeleton /> component should render the EbaySkeletonButton component 1`] = `
<div
  aria-label="Loading..."
  class="skeleton"
  role="img"
  style="width: 220px;"
>
  <div
    class="skeleton__button"
  />
</div>
`;

exports[`Render <EbaySkeleton /> component should render the EbaySkeletonButton with "as" prop component 1`] = `
<div
  aria-label="Loading..."
  class="skeleton"
  role="img"
  style="width: 220px;"
>
  <span
    class="skeleton__button"
  />
</div>
`;

exports[`Render <EbaySkeleton /> component should render the EbaySkeletonButton with size prop large component 1`] = `
<div
  aria-label="Loading..."
  class="skeleton"
  role="img"
  style="width: 220px;"
>
  <div
    class="skeleton__button skeleton__button--large"
  />
</div>
`;

exports[`Render <EbaySkeleton /> component should render the EbaySkeletonButton with size prop small component 1`] = `
<div
  aria-label="Loading..."
  class="skeleton"
  role="img"
  style="width: 220px;"
>
  <div
    class="skeleton__button skeleton__button--small"
  />
</div>
`;

exports[`Render <EbaySkeleton /> component should render the EbaySkeletonImage component 1`] = `
<div
  aria-label="Loading..."
  class="skeleton"
  role="img"
>
  <div
    class="skeleton__image"
    style="width: 220px; height: 220px;"
  />
</div>
`;

exports[`Render <EbaySkeleton /> component should render the EbaySkeletonImage with "as" prop component 1`] = `
<div
  aria-label="Loading..."
  class="skeleton"
  role="img"
>
  <span
    class="skeleton__image"
    style="width: 220px; height: 220px;"
  />
</div>
`;

exports[`Render <EbaySkeleton /> component should render the EbaySkeletonText component 1`] = `
<div
  aria-label="Loading..."
  class="skeleton"
  role="img"
  style="width: 220px;"
>
  <div
    class="skeleton__text"
  />
</div>
`;

exports[`Render <EbaySkeleton /> component should render the EbaySkeletonText with "as" prop component 1`] = `
<div
  aria-label="Loading..."
  class="skeleton"
  role="img"
  style="width: 220px;"
>
  <span
    class="skeleton__text"
  />
</div>
`;

exports[`Render <EbaySkeleton /> component should render the EbaySkeletonText with multiline prop component 1`] = `
<div
  aria-label="Loading..."
  class="skeleton"
  role="img"
  style="width: 220px;"
>
  <div
    class="skeleton__text skeleton__text--multiline"
  />
</div>
`;

exports[`Render <EbaySkeleton /> component should render the EbaySkeletonText with size prop component 1`] = `
<div
  aria-label="Loading..."
  class="skeleton"
  role="img"
  style="width: 220px;"
>
  <div
    class="skeleton__text skeleton__text--large"
  />
</div>
`;

exports[`Render <EbaySkeleton /> component should render the EbaySkeletonTextBox component 1`] = `
<div
  aria-label="Loading..."
  class="skeleton"
  role="img"
  style="width: 220px;"
>
  <div
    class="skeleton__textbox"
  />
</div>
`;

exports[`Render <EbaySkeleton /> component should render the EbaySkeletonTextBox with "as" prop component 1`] = `
<div
  aria-label="Loading..."
  class="skeleton"
  role="img"
  style="width: 220px;"
>
  <span
    class="skeleton__textbox"
  />
</div>
`;
